Dynamic Programming/0-1 Knapsack/Knapsack.playground/Contents.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+class KnapsackItem: NSObject {
+    var weight: Int
+    var value: Int
+    
+    init(weight: Int, value: Int) {
+        self.weight = weight
+        self.value = value
+    }
+}
+
+var capacityOfBag = Int()
+var knapsackItems = [KnapsackItem]()
+
+func knapsack() -> Int {
```

Comment:
This was somewhat confusing to read. Instead of having this free floating function for preparing the knapsack, let's create a class that models the knapsack. I've came up with this:

```
public final class Knapsack {
  public let capacity: Int

  // capacityOfBag would be a property of this class, passed in via the initializer
  public init(weightCapacity: Int, itemTypes: [KnapsackItem]) {
    capacity = weightCapacity

    // do something with the KnapsackItems
  }
}
```

I can see a few ways to model the solution. We could configure this Knapsack with the optimal `KnapsackItem` combination right in the initializer, or we can create a function that takes a set of `KnapsackItem` values and returns the optimal combination.

Let me know if that makes sense.